lib/os/json.c

```diff
@@ -302,6 +302,23 @@ static int obj_init(struct json_obj *json, char *data, size_t len)
 	return 0;
 }
 
+static int arr_init(struct json_obj *json, char *data, size_t len)
+{
+	struct token token;
```

Comment:
I'm not sure why `struct token token;` is tripping rule 5.7 when the same thing is done across multiple functions (obj_next, obj_init).
[Rule 5.7](https://docs.zephyrproject.org/latest/contribute/coding_guidelines/index.html): "A tag name shall be a unique identifier"

Reply:
@cfriedt Do you have any idea what might be causing this?

Reply:
Maybe it's because the variable is named the same as the struct? Same variable name across different functions should really not be an issue.

In any case, rewriting with `struct token tok` should fix it.

Reply:
Hey Martin thanks for the help, I was guessing that might be it. I found it strange that it didn't trigger a warning for the obj_next and obj_init function which do the same thing (Maybe the rule check is only for new code?).
I'll go ahead and update all occurrences of `struct token token` to `struct token tok`, really appreciate the help!

Reply:
Yep, it checks only the diff, AFAIK.